boa/src/exec/mod.rs

```diff
@@ -149,10 +149,10 @@ impl Interpreter {
             callable,
         );
 
-        let new_func = Object::function(func);
+        let mut new_func = Object::function(func);
 
+        new_func.prototype = function_prototype.clone();
```

Comment:
Here I see we create a function and then assign it a prototype. I would do this with a `set_prototype()` setter as I explained before, but if we are using this a lot, we could maybe have a constructor that accepts a prototype too.

Reply:
It is probably better for the constructor to take it, function objects should always have a prototype instance.